libdnf/repo/Repo.cpp

```diff
@@ -1344,11 +1345,18 @@ void Repo::Impl::attachLibsolvRepo(LibsolvRepo * libsolvRepo)
 
 void Repo::Impl::detachLibsolvRepo()
 {
-    libsolvRepo->appdata = nullptr;
-    if (--nrefs > 0)
+    attachLibsolvMutex.lock();
+    if (libsolvRepo) {
```

Comment:
Can you write comments on what happens in all `if` branches and why?
It's not clear at the first sight.